weblate/api/views.py

```diff
@@ -750,6 +753,143 @@ def destroy(self, request, *args, **kwargs):
         return Response(status=HTTP_204_NO_CONTENT)
 
 
+class GlossaryViewSet(WeblateViewSet, UpdateModelMixin, DestroyModelMixin):
+    """Translation projects API."""
+
+    queryset = Glossary.objects.none()
+    serializer_class = GlossarySerializer
+    lookup_field = "id"
+
+    def get_queryset(self):
+        return (
+            Glossary.objects.filter_access(self.request.user)
+            .prefetch_related("source_language")
+            .order_by("id")
+        )
+
+    def perm_check(self, request):
+        if not request.user.has_perm("project.edit"):
```

Comment:
It should be permission on the object level, it should be checked on Glossary.project.